onnx/defs/tensor/defs.cc

```diff
@@ -2482,6 +2482,197 @@ ONNX_OPERATOR_SET_SCHEMA(
           updateOutputShape(ctx, 0, {N, C, H_out, W_out});
         }));
 
+static const char* AffineGrid_ver20_doc = R"DOC(
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    AffineGrid,
+    20,
+    OpSchema()
+        .Attr(
+            "align_corners",
+            "if align_corners=1, consider -1 and 1 to refer to the centers of the corner pixels. "
+            "if align_corners=0, consider -1 and 1 to refer to the outer edge the corner pixels.",
+            AttributeProto::INT,
+            static_cast<int64_t>(0))
+        .Input(
+            0,
+            "theta",
+            "input batch of affine matrices with shape (N, 2, 3) for 2D or (N, 3, 4) for 3D",
+            "T1",
+            OpSchema::Single,
+            true,
+            1,
+            OpSchema::NonDifferentiable)
+        .Input(
+            1,
+            "size",
+            "the target output image size (N, C, H, W) for 2D or (N, C, D, H, W) for 3D",
+            "T2",
+            OpSchema::Single,
+            true,
+            1,
+            OpSchema::NonDifferentiable)
+        .Output(
+            0,
+            "grid",
+            "output tensor of shape (N, C, H, W, 2) of 2D sample coordinates or (N, C, D, H, W, 3) of 3D sample coordinates.",
+            "T1",
+            OpSchema::Single,
+            true,
+            1,
+            OpSchema::Differentiable)
+        .TypeConstraint(
+            "T1",
+            {"tensor(float16)", "tensor(float)", "tensor(double)"},
```

Comment:
Why not bfloat16 also? May be better to define and use some set like `all_float_types_ir4` (like [here](https://github.com/onnx/onnx/blob/a1e97a1ff257c7e3b1ecd62dc4872fa79059a275/onnx/defs/schema.h#L579) ... that will make ops follow similar pattern.